### DISCUSSION TITLE
What code populates the employees HTML table from the database?

### DISCUSSION THREAD

**Discussion by owlbrudder** [opensourcepos/opensourcepos#3806] [General]
I have been writing software commercially since 1980 and for eight years I was software development manager for a POS company.

To fill in some spare time, I am converting OSPOS to PHP8.x and CodeIgniter 4. While under development, I have called it OSPOS_HC to keep it visually distinct from the version 3.3.8 that I am running in parallel. When it is all working, it will be reverting back to the original name and all licencing will be as in the original - I am not doing this for money and I am not stealing any intellectual property  "8-).

Progress has been good, but I have run into a problem with the employees page. It should show at least one employee, John Doe, but it says there are no people to display.

I have searched everywhere I can think of for the code that populates the HTML table, but I can't see it. I know the data is massaged by Bootstrap, but where is the data coming from?

Any pointers will be greatly appreciated.

![OSPOS showing empty employees table](https://github.com/opensourcepos/opensourcepos/assets/20263907/42072ed3-3f9b-4d51-94d8-be7219d5b6bc)

**Comment by owlbrudder** [comment]
The solution was a series of steps to track down HTTP 500 problems which all came down to the way CI4 handles database queries. My initial rewrite was not absolutely accurate.

**Reply by jekkos** [reply]
HI, quite some work in this area has already been done in the ci4-upgrade branch. You can check the open issue for points you can help the project with, https://github.com/opensourcepos/opensourcepos/issues/3584

**Reply by owlbrudder** [reply]
I will be delighted if any of my work is useful to the wider community. I am doing my conversion as an hobby and will complete it, but I am sure the original team will be doing a better job. Thanks for the suggestion.
